### ISSUE TITLE
cli shouldn't mess with sys.argv

### ISSUE THREAD

**Issue by davidism** [pallets/flask#3022]
Just merged https://github.com/pallets/werkzeug/pull/1416 so Werkzeug's reloader detects `python -m module` correctly. All of `flask.cli.main()` can be replaced with:

```python
def main(as_module=False):
    cli.main(prog_name="python -m flask" if as_module else None)
```

Need to wait until Werkzeug 0.15 has been out for a bit before we can bump Flask's min version.

**Comment by seawolf42** [comment]
picking up this task

**Comment by EtiennePelletier** [comment]
PR https://github.com/pallets/flask/pull/3181 updates Werkzeug to >=0.15, so once merged I'll take care of this ticket.

**Comment by seawolf42** [comment]
@EtiennePelletier if you're going to do it, you don't need to wait for the other to be merged, but you should reference that PR (using the # reference like above) in the new PR so that they see the dependency when they review this one.

**Comment by EtiennePelletier** [comment]
Alright! I just wanted to avoid bumping Werkzeug's version in setup.py again since master is currently referencing 0.14. I'll leave it as is and explicitly mention it in the PR.
